Updates for Katello 4.10.0 - Release notes (#2481)
theforeman/foreman-documentation: guides/doc-Release_Notes/topics/katello-4.10.0.adoc

= Katello 4.10.0

A full list of changes is available on https://projects.theforeman.org/issues?set_filter=1&sort=id%3Adesc&status_id=closed&f%5B%5D=cf_12&op%5Bcf_12%5D=%3D&v%5Bcf_12%5D%5B%5D=1687[Redmine]

== Katello

> * Incorrect aria-label and component id in the alternate content source modal for editing credentials certificates - https://projects.theforeman.org/issues/36512[#36512]

Reviewer: ```suggestion
* Incorrect aria-label and component ID in the alternate content source modal for editing credentials certificates - https://projects.theforeman.org/issues/36512[#36512]
```
Even below.

Author: Thanks! Pushed these edits and other typos where / was present for some characters

* ID for activation keys on register host page is incorrect - https://projects.theforeman.org/issues/36489[#36489]

=== API

* Remove deprecated docker_tags_whitelist - https://projects.theforeman.org/issues/36695[#36695]
* Remove katello-agent API endpoints - https://projects.theforeman.org/issues/36665[#36665]

=== Activation Key

* Add activation key details top bar - https://projects.theforeman.org/issues/36576[#36576]
* Environment and Content View info is not visible on the Associations Content Host page for Activation keys - https://projects.theforeman.org/issues/36501[#36501]

=== Alternate Content Sources

> * ACS bulk refresh through API silently sanitizes input ids - https://projects.theforeman.org/issues/36634[#36634]

Reviewer: ```suggestion
* ACS bulk refresh through API silently sanitizes input IDs - https://projects.theforeman.org/issues/36634[#36634]
```

* Custom ACS path help text is missing file:// - https://projects.theforeman.org/issues/36631[#36631]
* ACS error message contains duplicate words - https://projects.theforeman.org/issues/36590[#36590]
* Add option --all to "hammer alternate-content-source refresh" command to refresh all ACS's - https://projects.theforeman.org/issues/36589[#36589]

> * Incorrect aria-label and component id in the alternate content source modal for editing Products - https://projects.theforeman.org/issues/36514[#36514]

Reviewer: ```suggestion
* Incorrect aria-label and component ID in the alternate content source modal for editing Products - https://projects.theforeman.org/issues/36514[#36514]
```

> * Incorrect aria-label and component id in the alternate content source modal for editing smart proxies - https://projects.theforeman.org/issues/36513[#36513]

Reviewer: ```suggestion
* Incorrect aria-label and component ID in the alternate content source modal for editing smart proxies - https://projects.theforeman.org/issues/36513[#36513]
```

* Incorrect aria-label in the alternate content source drawer - https://projects.theforeman.org/issues/36511[#36511]

> * Incorrect aria-label and component id in the alternate content source modal for editing credentials - https://projects.theforeman.org/issues/36510[#36510]

Reviewer: ```suggestion
* Incorrect aria-label and component ID in the alternate content source modal for editing credentials - https://projects.theforeman.org/issues/36510[#36510]
```

> * \'Remove orphans\' task fails on DeleteOrphanAlternateContentSources step - https://projects.theforeman.org/issues/36461[#36461]

Reviewer 2: The script generates incorrect escaping. You can see it renders it incorrectly if you go to https://theforeman-foreman-documentation-preview-pr-2481.surge.sh/3.8/Release_Notes/index-katello.html#_alternate_content_sources

In the past I've always fixed these manually by reviewing the rendered document. https://github.com/theforeman/foreman-documentation/pull/2427 aims to solve it properly, but since that's still in review I'd suggest to also manually review all escaping.

* Simplified ACS update fails to remove products if product has any empty URL repos - https://projects.theforeman.org/issues/36221[#36221]

=== Content Views

* Content View filter creation/edit button should say "Save" instead of "Edit rule" - https://projects.theforeman.org/issues/36687[#36687]
* undefined method '#single_content_view' for Katello::Host::ContentFacet::Jail (Katello::Host::ContentFacet) (Safemode::NoMethodError) - https://projects.theforeman.org/issues/36684[#36684]
* Adding a CV to a CCV lists CV versions disorderly - https://projects.theforeman.org/issues/36679[#36679]
* Solve dependencies in settings is a dead setting - https://projects.theforeman.org/issues/36654[#36654]
* "Add content view" window and "Update version" window should display content view version, description and publishing date - https://projects.theforeman.org/issues/36648[#36648]
* update event is not created after updating CV name - https://projects.theforeman.org/issues/36628[#36628]

> * Re-synchronizing the repository whose sync had failed earlier, the Content View does not display new upgradable packages that are available. - https://projects.theforeman.org/issues/36622[#36622]

Reviewer: ```suggestion
* Re-synchronizing the repository whose sync had failed earlier, the Content View does not display new upgradable packages that are available - https://projects.theforeman.org/issues/36622[#36622]
```

* Content View API lists same Environment Name and Label despite name changes - https://projects.theforeman.org/issues/36598[#36598]

> * Hammer should treat indeterminate needs_publish as publish_needed. - https://projects.theforeman.org/issues/36581[#36581]

Reviewer: ```suggestion
* Hammer should treat indeterminate needs_publish as publish_needed - https://projects.theforeman.org/issues/36581[#36581]
```

* Content view dependency solving should be tracked for needs_publish - https://projects.theforeman.org/issues/36580[#36580]

> * Filter gets applied to all the repository upon removal of repository for which the filter was created. - https://projects.theforeman.org/issues/36577[#36577]

Reviewer: ```suggestion
* Filter gets applied to all the repositories upon removal of the repository for which the filter was created. - https://projects.theforeman.org/issues/36577[#36577]
```

* CV page needs refresh to get the current filters state - https://projects.theforeman.org/issues/36529[#36529]

> * Content View comparison - RPM packages search missing auto completion - https://projects.theforeman.org/issues/36516[#36516]

Reviewer: ```suggestion
* Content View comparison - RPM packages search missing auto-completion - https://projects.theforeman.org/issues/36516[#36516]
```

> * Content view Promote always warns for 'Force promotion' even if correct path is followed. - https://projects.theforeman.org/issues/36515[#36515]

Reviewer: ```suggestion
* Content view Promote always warns for 'Force promotion' even if the correct path is followed. - https://projects.theforeman.org/issues/36515[#36515]
```

> * Content view Promote always warns for 'Force promotion' even if the correct path is followed. - https://projects.theforeman.org/issues/36515[#36515]

Reviewer: ```suggestion
* Content view Promote always warns for 'Force promotion' even if the correct path is followed - https://projects.theforeman.org/issues/36515[#36515]
```

* Link for container count on CV page redirects to an invalid page - https://projects.theforeman.org/issues/36474[#36474]
* Unable to disable import_only flag in Satellite UI when set on Content Views - https://projects.theforeman.org/issues/36459[#36459]

> * Content view publish with filters is getting failed with error "Could not find the following content units:" - https://projects.theforeman.org/issues/36334[#36334]

Reviewer: ```suggestion
* Content view publish with filters is getting failed with the error "Could not find the following content units:" - https://projects.theforeman.org/issues/36334[#36334]
```

* Wrong listing of Content Views which contain Files - https://projects.theforeman.org/issues/36288[#36288]

=== Errata Management

* Allow installable errata count methods - https://projects.theforeman.org/issues/36506[#36506]

=== Foreman Proxy Content

* Restore n-1 smart proxy sync support for pulpcore 3.28 / 3.22 - https://projects.theforeman.org/issues/36686[#36686]
* Capsule Content view's 'Last published' field is confusing - https://projects.theforeman.org/issues/36629[#36629]
* The "POST /katello/api/capsules/:id/reclaim_space" endpoint is wrong - https://projects.theforeman.org/issues/36545[#36545]
* Optimized capsule sync doesn't sync recently published/promoted docker repositories - https://projects.theforeman.org/issues/36523[#36523]
* syncing a capsule fails with ActiveRecord::RecordNotFound Couldn't find SmartProxy with 'id'=2 - https://projects.theforeman.org/issues/36520[#36520]
* External capsule is auto-synced on CV promotion regardless foreman_proxy_content_auto_sync settings - https://projects.theforeman.org/issues/36442[#36442]

> * Capsule redundantly synces *-Export-Library repos - https://projects.theforeman.org/issues/36436[#36436]

Reviewer: ```suggestion
* Capsule redundantly syncs *-Export-Library repos - https://projects.theforeman.org/issues/36436[#36436]
```

> * Orphan cleanup runs fine but does not clears anything from /var/lib/pulp/media/artifact of Red Hat Capsule 6.10 - https://projects.theforeman.org/issues/36390[#36390]

Reviewer: ```suggestion
* Orphan cleanup runs fine but does not clear anything from /var/lib/pulp/media/artifact of Red Hat Capsule 6.10 - https://projects.theforeman.org/issues/36390[#36390]
```

=== Hammer

* hammer host info no longer shows content view and lifecycle environment - https://projects.theforeman.org/issues/36401[#36401]

=== Hosts

> * Discovery Provisioning fails as the sync media is getting diminished from the host page even though its synced. - https://projects.theforeman.org/issues/36721[#36721]

Reviewer: ```suggestion
* Discovery Provisioning fails as the sync media is getting diminished from the host page even though it's synced. - https://projects.theforeman.org/issues/36721[#36721]
```

> * Discovery Provisioning fails as the sync media is getting diminished from the host page even though it's synced. - https://projects.theforeman.org/issues/36721[#36721]

Reviewer: ```suggestion
* Discovery Provisioning fails as the sync media is getting diminished from the host page even though it's synced - https://projects.theforeman.org/issues/36721[#36721]
```

* Can not re-register with --force after deleting consumer certs against satellite614 - https://projects.theforeman.org/issues/36674[#36674]
* Hammer accepts non-existent LCE in host update - https://projects.theforeman.org/issues/36667[#36667]
* New host details UI - Remove katello-agent code - https://projects.theforeman.org/issues/36664[#36664]
* Change host content source warning implies overall failure when it should instead tell you that step 1 of 2 is complete - https://projects.theforeman.org/issues/36623[#36623]
* Change content source page never shows job invocation link - https://projects.theforeman.org/issues/36621[#36621]
* Errors due to lack of safe navigation when you try to customize a discovered host - https://projects.theforeman.org/issues/36608[#36608]

> * Error when autoprovision/provision for a discovered host - https://projects.theforeman.org/issues/36601[#36601]

Reviewer: ```suggestion
* Error when auto-provision/provision for a discovered host - https://projects.theforeman.org/issues/36601[#36601]
```

* Rename 'Change content source' job template so it's less confusing - https://projects.theforeman.org/issues/36597[#36597]
* undefined method 'each' for #&lt;Katello::ContentViewEnvironment when running hammer host subscription register - https://projects.theforeman.org/issues/36524[#36524]
* undefined method 'content_view=' for #&lt;Katello::Host::ContentFacet:0x00007fc530855ac8&gt; - https://projects.theforeman.org/issues/36504[#36504]
* Editing a host results in an error "content_view_id and lifecycle_environment_id must be provided together" - https://projects.theforeman.org/issues/36498[#36498]

> * User with "Register Hosts" role ignores all the setup options. - https://projects.theforeman.org/issues/36484[#36484]

Reviewer: ```suggestion
* User with "Register Hosts" role ignores all the setup options - https://projects.theforeman.org/issues/36484[#36484]
```

* Can't add hostgroup to new host - https://projects.theforeman.org/issues/36462[#36462]
* Host details UI, Repository sets table contains empty column header - https://projects.theforeman.org/issues/36445[#36445]
* hammer host update fails with "unknown attribute ‘content_view_id’ for Katello::Host::ContentFacet" when you pass a content view / LCE - https://projects.theforeman.org/issues/36440[#36440]
* Arch restriction label missing from Repository sets for repos without URL - https://projects.theforeman.org/issues/36430[#36430]

> * Installable update links on the host page still link to the old content host detail page. - https://projects.theforeman.org/issues/36254[#36254]

Reviewer: ```suggestion
* Installable update links on the host page still link to the old content host detail page - https://projects.theforeman.org/issues/36254[#36254]
```

> * Installable updates links on the hosts page still link to the old content host detail page. - https://projects.theforeman.org/issues/36254[#36254]

Reviewer: ```suggestion
* Installable update links on the host page still link to the old content host detail page. - https://projects.theforeman.org/issues/36254[#36254]
```

* Remove the setting 'Use remote execution by default' - https://projects.theforeman.org/issues/36083[#36083]

=== Inter Server Sync

* hammer content import fails with undefined method 'substitutor' for nil:NilClass during import content if product being imported is not covered by subscriptions on the manifest - https://projects.theforeman.org/issues/36521[#36521]
* Unable to enable any repository in network sync - https://projects.theforeman.org/issues/36482[#36482]
* Exporting repositories that have architecture restrictions results in bogus data that can't be properly imported - https://projects.theforeman.org/issues/36477[#36477]
* Can't update the redhat_repository_url without changing the cdn_configuration to custom_cdn - https://projects.theforeman.org/issues/36463[#36463]

=== Localization

* split out mo file and po file adding to git - https://projects.theforeman.org/issues/36444[#36444]

=== Organizations and Locations

* edit_organization permissions needed on upstream satellite  - https://projects.theforeman.org/issues/36503[#36503]

=== Repositories

* Use pulp_deb 3.0.0 - https://projects.theforeman.org/issues/36728[#36728]

> * hammer- allow user to run Verify Content Checksum, on container repositories. - https://projects.theforeman.org/issues/36625[#36625]

Reviewer: ```suggestion
* hammer- allow user to run Verify Content Checksum, on container repositories - https://projects.theforeman.org/issues/36625[#36625]
```

* Repository details page shouldn't say 'enabled by default' - https://projects.theforeman.org/issues/36593[#36593]
* Repository export fails with Error "Validation failed: Relative path is too long - https://projects.theforeman.org/issues/36584[#36584]
* 4.9 is slow to sync and index repositories - https://projects.theforeman.org/issues/36563[#36563]
* Upgrade to Katello 4.5 can fail if some on_demand repositories have checksum_type set - https://projects.theforeman.org/issues/36562[#36562]
* Hide option to delete content from Redhat repos - https://projects.theforeman.org/issues/36554[#36554]
* Remove deprecated & not working API endpoints from APIdoc - https://projects.theforeman.org/issues/36530[#36530]

> * Optimize DockerMetaTag query and CV version deletion to run a single invocation of the method. - https://projects.theforeman.org/issues/36500[#36500]

Reviewer: ```suggestion
* Optimize DockerMetaTag query and CV version deletion to run a single invocation of the method - https://projects.theforeman.org/issues/36500[#36500]
```

* Bump recommended Red Hat repos for 6.14 - https://projects.theforeman.org/issues/36485[#36485]
* Prevent regenerating metadata for repositories that use complete mirroring - https://projects.theforeman.org/issues/36453[#36453]

> * Updating ignorable_content should not trigger pulp updates cause there is nothing to update in pulp. - https://projects.theforeman.org/issues/36428[#36428]

Reviewer: ```suggestion
* Updating ignorable_content should not trigger pulp updates cause there is nothing to update in pulp - https://projects.theforeman.org/issues/36428[#36428]
```

=== Subscriptions

* Reasons for not deleting the manifest don't apply with SCA enabled - https://projects.theforeman.org/issues/36604[#36604]
* A failed "Actions::Katello::Host::Hypervisors" task lacks identifying information - https://projects.theforeman.org/issues/36599[#36599]
* Fix the Documentation link in the Manifest history tab - https://projects.theforeman.org/issues/36272[#36272]

=== Tests

> * Pin ostree binding on nightly for tests to pass. - https://projects.theforeman.org/issues/36586[#36586]

Reviewer: ```suggestion
* Pin ostree binding on nightly for tests to pass - https://projects.theforeman.org/issues/36586[#36586]
```

* Add option --all to "hammer alternate-content-source refresh" command to refresh all ACS\'s - https://projects.theforeman.org/issues/36589[#36589]
* Incorrect aria-label and component id in the alternate content source modal for editing Products - https://projects.theforeman.org/issues/36514[#36514]
* Incorrect aria-label and component id in the alternate content source modal for editing smart proxies - https://projects.theforeman.org/issues/36513[#36513]
* Incorrect aria-label in the alternate content source drawer - https://projects.theforeman.org/issues/36511[#36511]
* Incorrect aria-label and component id in the alternate content source modal for editing credentials - https://projects.theforeman.org/issues/36510[#36510]
* \'Remove orphans\' task fails on DeleteOrphanAlternateContentSources step - https://projects.theforeman.org/issues/36461[#36461]
* Simplified ACS update fails to remove products if product has any empty URL repos - https://projects.theforeman.org/issues/36221[#36221]

=== Client/Agent

* tasks Actions::Katello::BulkAgentAction without any sub-plans and stuck in running/pending - https://projects.theforeman.org/issues/36528[#36528]

> * Add UI banners warning users about katello-agent removal - https://projects.theforeman.org/issues/36526[#36526]

Reviewer 3: These two are irrelevant now that katello-agent is gone. May think about omitting these items (not the code, just this text) just to avoid confusion.
